specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/automations.json

```diff
@@ -497,7 +497,6 @@
     "AutomationActionLogicAppInput": {
       "type": "object",
       "description": "The logic app action that should be triggered.",
-      "x-ms-discriminator-value": "LogicApp",
```

Comment:
Is `AutomationActionLogicAppInput` never sent across the wire?

Reply:
Same question applies to the event hub types.

Reply:
AutomationActionLogicApp is kind of an "abstract" model which is derived to two models: AutomationActionLogicAppInput - which is being **sent** for PUT requests, and AutomationActionLogicAppOutput - which is the one that is **returned** for GET and PUT. Actually, there is the "uri" field which is not allowed to be returned for security reasons.
Same approach is for EventHub.

I had tagged the 3 models (abstract + input + output) with the x-ms-discriminator-value property, for the same value ("LogicApp"). Apparently this caused the GO SDK to fail.
I am wondering how the following schema should be presented in swagger though:

(copied from examples)
**PUT Request**:
"actions": [
          {
            "logicAppResourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
            "actionType": "LogicApp",
            "uri": "https://exampleTriggerUri1.com"
          }
        ]
**Response**:
"actions": [
            {
              "logicAppResourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
              "actionType": "LogicApp"
            }
          ]

Reply:
If `AutomationActionLogicApp` is a "base class" that is never sent/received then I believe that it should omit the `x-ms-discriminator-value` and that each derived type should define its own `x-ms-discriminator-value` value.  I am by no means an expert in this area though, @fearthecowboy can you please advise?